### ISSUE TITLE
NODE_ENV is undefined

### ISSUE THREAD

**Issue by PointSingularity** [nrwl/nx#16960]
### Current Behavior

Creating a new standalone Node application with NX 16.1.0 and console logging ```process.env.NODE_ENV``` gives back undefined,
even though it was always defined in versions 15 anb below.

Using --prod during build also doesn't change anything.

### Expected Behavior

process.env.NODE_ENV='development' during development.
process.env.NODE_ENV='production' after using build to do a production build.

### Steps to Reproduce

1. Create a standalone Node application with 16.1.0
2. Put console.log(process.env.NODE_ENV) in main.ts and run it
3. Additionally build the application and run the main.js from the dist folder.

### Nx Report

```shell
Node   : 18.15.0
   OS     : win32 x64
   npm    : 9.6.3
   Hasher : Native

   nx                 : 16.1.0
   @nx/js             : 16.1.0
   @nx/jest           : 16.1.0
   @nx/linter         : 16.1.0
   @nx/workspace      : 16.1.0
   @nx/devkit         : 16.1.0
   @nx/esbuild        : 16.1.0
   @nx/eslint-plugin  : 16.1.0
   @nx/node           : 16.1.0
   @nrwl/tao          : 16.1.0
   typescript         : 5.0.4
```
```


### Failure Logs

_No response_

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

**Comment by renambot** [comment]
same problem.

**Comment by oben01** [comment]
same problem also

**Comment by janeklb** [comment]
temp workaround 

```ts
// in your webpack.config.ts

// workaround for undefined NODE_ENV
const withFixedNodeEnv: () => NxWebpackPlugin =
  () =>
  (config, { context }) => {
    if (config.mode && config.mode !== 'none') {
      return config;
    }
    if (context.configurationName === 'production') {
      config.mode = 'production';
    } else {
      config.mode = 'development';
    }
    return config;
  };

export default composePlugins(withNx(), withFixedNodeEnv(), (config) => {
  // Update the webpack config as needed here.
  // e.g. config.plugins.push(new MyPlugin())
  // For more information on webpack config and Nx see:
  // https://nx.dev/packages/webpack/documents/webpack-config-setup
  return config;
});
```

**Comment by PointSingularity** [comment]
@ndcunningham Thank you for fixing this for esbuild! Sadly, this is also present for webpack too, so I don't think it's fully resolved yet.

**Comment by bojanbass** [comment]
I'm not sure the fix that was merged has anything to do with `@nx/js:node` serve and `@nx/webpack:webpack` executors, as one of those is causing this error. I think this bug report has to bug reopened.

**Comment by bojanbass** [comment]
A temporary workaround from @janeklb does not work in my case. I have managed to run it properly using `.env` configs per target and configuration:
- `.env.serve.development` (NODE_ENV=development)
- `.env.serve.production` (NODE_ENV=production)

This way, serving the app with the `--prod` flag or without loads the correct .env file. But still, this is quite a hack and a proper solution should be implemented.

I did some debugging and my closest bet is that this PR broke the previously working functionality: https://github.com/nrwl/nx/commit/dc7122984e5a6be7c9aa28fd492b8ad310f7a709

I suppose executing a `buildTarget` with `npx nx run` [command](https://github.com/nrwl/nx/blob/master/packages/js/src/executors/node/node.impl.ts#L93-L96) looses `process.env.NODE_ENV` value. Another strange thing is that when triggering the `serve` target and inspecting the `build` target which uses `@nx/webpack:webpack` under the hood it does not output any `console.log` statement. Still, if I trigger the `build` target directly, it does. In previous NX versions, this was also working correctly. 

@jaysoo any idea on that one?
